### PR TITLE
style: Make eslint stricter, fix issues that came up

### PR DESCRIPTION
**Pull request by robbie-c**
## Problem

Just code style.

## Changes

In eslintrc:
* Added eslint:recommended to `extends`
* Added jest to `plugins`
* Disabled some rules which didn't seem helpful
* Tweaked env and globals

Elsewhere
* Ran eslint with --fix
* Mechnically changed anything that remained

Some specifics to draw attention to:
* no-useless-escape - many regexs has escapes in character classes that didn't need to be there, I just let eslint make these changes
* no-empty-pattern - in a few places we used object destrucuring syntax without actually pulling anything out, I either replaced this with `_` to signal an unused argument, or removed the argument where possible
* no-irregular-whitespace - I just disabled the rule as I was too afraid to touch this as there's no easy find&replace to run, and doing it manually felt like it would be error-prone. It would be good to fix this later, as there were a few places we used the "&nbsp;" (not a normal space) unicode character rather than writing out `&nbsp;`. This shows up clearly in some editors but not in e.g. github, see below
Screenshot 2023-08-18 at 14 07 49
* One regex triggers a security warning, I haven't fixed this as part of this PR.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?
Relied on our existing test suite as I am not adding anything new

### CODE REVIEW

File: frontend/src/lib/constants.tsx [hunk 1]
```diff
@@ -228,7 +228,6 @@ export const UPGRADE_LINK = (cloud?: boolean): { url: string; target?: '_blank'
     cloud ? { url: urls.organizationBilling() } : { url: 'https://posthog.com/pricing', target: '_blank' }
 
 export const DOMAIN_REGEX = /^([a-z0-9]+(-[a-z0-9]+)*\.)+[a-z]{2,}$/
-export const SECURE_URL_REGEX = /^(?:http(s)?:\/\/)[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:\/?#[\]@!\$&'\(\)\*\+,;=.]+$/gi
```
Comment (robbie-c) [inline frontend/src/lib/constants.tsx:231]: This is the regex I replaced with a proper URL parser. It accepts both http and https URLS, though I'm not sure if this was intended given the name.

File: frontend/src/lib/constants.tsx [hunk 2]
```diff
@@ -228,7 +228,7 @@
...
+export const SECURE_URL_REGEX = /^(?:http(s)?:\/\/)[\w.-]+(?:\.[\w.-]+)+[\w\-._~:/?#[\]@!$&'()*+,;=]+$/gi
```
[the remark this answers is not in the document]
Comment (robbie-c) [reply frontend/src/lib/constants.tsx:232]: I'm ignoring this in this PR, this problem isn't new, and I don't want to increase the scope

File: .eslintrc.js [hunk 3]
```diff
@@ -171,8 +177,23 @@ module.exports = {
                 ],
             },
         ],
+        'no-constant-condition': 0,
```
Comment (Twixes) [inline .eslintrc.js:180]: Why's this one disabled?

**Review by Twixes — APPROVED** [review]
🧹

File: .eslintrc.js (hunk 3 above)
Comment (Twixes) [reply .eslintrc.js:180]: Would maybe individually ignore those lines, but that's fine

File: package.json [hunk 4]
```diff
@@ -28,8 +28,8 @@
         "test:visual-regression:legacy:ci:verify": "playwright test",
         "test:visual-regression:stories": "docker compose -f docker-compose.playwright.yml run --rm -it --build playwright pnpm test:visual-regression:stories:docker",
         "test:visual-regression:stories:docker": "NODE_OPTIONS=--max-old-space-size=6144 test-storybook -u --no-index-json --browsers chromium webkit --url http://host.docker.internal:6006",
-        "test:visual-regression:stories:ci:update": "test-storybook -u --no-index-json --maxWorkers=2",
-        "test:visual-regression:stories:ci:verify": "test-storybook --ci --no-index-json --maxWorkers=2",
+        "test:visual-regression:stories:ci:update": "test-storybook -u --no-index-json --maxWorkers=4",
+        "test:visual-regression:stories:ci:verify": "test-storybook --ci --no-index-json --maxWorkers=4",
```
[the remark this answers is not in the document]
Comment (robbie-c) [reply package.json:32]: CC @Twixes

[the remark this answers is not in the document]
Comment (pauldambra) [reply package.json:32]: Was presumably when we moved to buildjet workers with all their swooshy CPUs 🤣 

Turning the dial back down to 2 seemed to fix things now we're back on GitHub's runners... We could lean on concurrency now that we have more runners and shard the tests more finely 🤔

**Pull request merged**